src/context/authContext.js

```diff
@@ -1,38 +1,38 @@
 import { createContext, useContext, useReducer, useEffect, useState } from "react";
 import axios from "axios";
 
-
 export const AuthContext = createContext();
 
 export const useAuth = () => useContext(AuthContext);
 
 export const AuthProvider = ({ children }) => {
-
     const [user, setUser] = useState(null);
     const [error, setError] = useState({ errorExists: false, errorMessage: "" })
-
+    const [encodedToken, setEncodedToken] = useState(localStorage.getItem("token"));
     const SignupUser = async (email, password) => {
         setError({
             errorExists: false,
             errorMessage: ""
         })
         try {
-
             const userResponse = await axios.post(`/api/auth/signup`, JSON.stringify({
                 email: email,
                 password: password,
             }));
             if (userResponse.statusText === "Created") {
                 localStorage.setItem("token", JSON.stringify(userResponse.data.encodedToken));
+                setEncodedToken(userResponse.data.encodedToken);
                 setUser(userResponse.data.createdUser);
+                return user;
             }
         }
         catch (e) {
             console.log("Found error: ", e);
-            setError({
-                errorExists: true,
-                errorMessage: e
-            })
+            // setError({
+            //     errorExists: true,
+            //     errorMessage: e
+            // })
+            alert("Credentials invalid. Please sign-up if you don't already have an account.");
```

Comment:
Commented code can be removed if not required anymore.

Reply:
done